### PR TITLE
Make the show/hide layer buttons work (#118)

### PR DESCRIPTION
**Pull request by akshay1992kalbhor**
@Keavon @TrueDoctor Any suggestions/changes?

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/127)

### CODE REVIEW

File: core/document/src/document.rs [hunk 1]
```diff
@@ -300,6 +300,14 @@ impl Document {
 				let children = self.layer_panel(path.as_slice())?;
 				Some(vec![DocumentResponse::DocumentChanged, DocumentResponse::ExpandFolder { path, children }])
 			}
+			Operation::ToggleVisibility { path } => {
+				let _ = self.layer_mut(&path).map(|x| {
```
Comment (Keavon) [inline core/document/src/document.rs:304]: Please rename `x` to `layer`.

File: core/document/src/document.rs [hunk 2]
```diff
@@ -300,6 +300,14 @@ impl Document {
 				let children = self.layer_panel(path.as_slice())?;
 				Some(vec![DocumentResponse::DocumentChanged, DocumentResponse::ExpandFolder { path, children }])
 			}
+			Operation::ToggleVisibility { path } => {
```
Comment (Keavon) [inline core/document/src/document.rs:303]: @TrueDoctor did we decide we generally want to do `set` or `toggle`?

Comment (TrueDoctor) [reply core/document/src/document.rs:303]: For now toggle

**Comment by akshay1992kalbhor** [comment]
I have made the necessary changes. Is there a way to update this PR with the changes? Or do I need to make a new one?

**Review by Keavon — APPROVED** [review]

File: core/document/src/document.rs [hunk 3]
```diff
@@ -300,6 +300,14 @@ impl Document {
...
+				let _ = self.layer_mut(&path).map(|layer| {
+					layer.visible = !layer.visible;
+					layer.cache_dirty = true;
+				});
+				let children = self.layer_panel(&[])?;
```
Comment (TrueDoctor) [inline core/document/src/document.rs:308]: you should either update the folder the current layer resides (for that you can use `children = self.layer_panel(path.as_slice()[..path.len()-1])?`)
or you need to add a separate response for the vis toggle

Comment (akshay1992kalbhor) [reply core/document/src/document.rs:308]: Should `children = self.layer_panel(path.as_slice()[..path.len()-1])?` replace line 308?

Comment (TrueDoctor) [reply core/document/src/document.rs:308]: you still need the `let` but yeah, you can try and see if it works

**Comment by Keavon** [comment]
Closes #118

**Pull request merged**